quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/oas/OASModelFilter.java

```diff
@@ -0,0 +1,62 @@
+package org.keycloak.quarkus.runtime.oas;
+
+import org.eclipse.microprofile.openapi.OASFactory;
+import org.eclipse.microprofile.openapi.OASFilter;
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.models.PathItem;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class OASModelFilter implements OASFilter {
```

Comment:
```suggestion

@OpenApiFilter(OpenApiFilter.RunStage.BUILD) 
public class OASModelFilter implements OASFilter {
```

@robson90 Would it be possible to run the filter in build time? AFAIK we do not expect/support any dynamic endpoint registration for now. 

https://quarkus.io/guides/openapi-swaggerui#enhancing-the-openapi-schema-with-filters

---------------------------------

Also, the Swagger UI might look better: 
<img width="2209" height="1162" alt="Screenshot From 2025-08-14 11-55-00" src="https://github.com/user-attachments/assets/05d6dc97-94d0-4b85-9d43-056592d60e06" />

- Do not show the empty categories (Roles, Role Mapper,...)
- Have these endpoints categorized based on the class (not `default`) - name of the classes might be like "Realms (v2)", "Clients (v2)"

Reply:
I can refine the Filter. Ive found this: https://github.com/quarkusio/quarkus/issues/40007#issuecomment-2069764206
That should be applicable for our use case, for filtering everything, that is not V2.

I will check out the Annotation.

Reply:
With some filtering it looks now like this:
<img width="1342" height="1489" alt="Screenshot 2025-08-15 at 09 45 52" src="https://github.com/user-attachments/assets/6e232102-8e3f-4f29-9a5c-c7dbd50a3838" />

@mabartos WDYT?

Reply:
@robson90 Nice! IMHO it looks much better. Are many changes needed to achieve this?

Reply:
Not really, just some filtering and setting a new Tag for the ClientsAPI.

I will push the changes later, and keep you posted

Reply:
Great, thanks! Did you have a chance to include the `@OpenApiFilter(OpenApiFilter.RunStage.BUILD)` annotation as well?